server/channels/db/migrations/mysql/000121_drop_threads_teamid.up.sql

```diff
@@ -0,0 +1,20 @@
+SET @preparedStatement = (SELECT IF(
+    (
+	SELECT row_format FROM INFORMATION_SCHEMA.TABLES
+	WHERE table_name = 'Threads'
+		AND table_schema = DATABASE()
+	) != 'Compact'
```

Comment:
Can you elaborate on this part?

Reply:
I'm wondering if we should skip this check and "let it fail" if we can't apply it instantly? Otherwise the user ends up with a working, but blocking migration (violating our guarantees).

Reply:
I also added this to the migration guide: https://docs.mattermost.com/deploy/postgres-migration.html#artifacts-may-remain-from-previous-configurations-versions cc @lieut-data

Reply:
> It will only run if it's a non-blocking migration (which should be the ideal and expected scenario).

To clarify, if a customer had compressed row formats, would it be expected that some parts of the migration are silently skipped? I wonder if an error on failing to do something is a more ideal scenario?

Reply:
I see, we can log that separately for sure but the migration can't fail (it would stop application from starting). I agree that this is not an ideal scenario but also thinking whether we should allow using different row formats.

Reply:
> the migration can't fail (it would stop application from starting)

Why can't the migration fail? If we can't migrate for some reason, is failure a preventative safeguard against starting the application with an unsupported migration?

Reply:
Yes, we don't have a graceful recovery other than rolling back to older version/schema.

Reply:
> Yes, we don't have a graceful recovery other than rolling back to older version/schema.

Right, but that's the correct thing to do if a migration fails, right? I'm currently thinking that we should never allow a "silent failure".

Reply:
I see, this is what the compromise we were thinking by simply ignoring the fact that column may exist. @lieut-data we can log this by adding a new value to the `Systems` table to indicate that this was skipped. I wouldn't log this somewhere else as it can get lost and would not be easy to figure out if it was skipped (ie silently failed). WDYT? Would it make sense to you?